pkg/controller/nodes/task/taskexec_context.go

```diff
@@ -43,6 +44,19 @@ func (te taskExecutionID) GetGeneratedName() string {
 	return te.execName
 }
 
+func (te taskExecutionID) GetGeneratedNameWith(minLength, maxLength int) (string, error) {
```

Comment:
Does it makes sense to use the GetGeneratedName function rather than te.execName in this function? Then an implementation can override the GetGeneratedName function (like I'm exploring in the maptask refactor) and this function will still work.

Reply:
If I understand you correctly, then I don't think this will work... Go doesn't have inheritance. It rather supports extending types.. does [this example](https://go.dev/play/p/jiCZ6hqmsuC) show what you have in mind?